Experiment automation/(LUMICKS) Reference kit automation script/Catch_beads_DNA_record_Kymo_for_LacI_version_1.py

```diff
@@ -0,0 +1,318 @@
+    n = 0
+    while match_score1.latest_value < match_threshold or match_score2.latest_value < match_threshold:
+        """Drop beads that do not fulfill the template"""
+        if 0 < match_score1.latest_value < match_threshold:
+            bl.mirror1.clear()
+        if 0 < match_score2.latest_value < match_threshold:
+            bl.mirror2.clear()
+        if n > 50:
+            stop_flow()
+            raise Exception("Need more than 750 seconds to catch beads, stop script")
+        """If it's taking too long, maybe something is stuck in the trap. Clear both traps."""
+        if time.time() - start_time > 15:
+            bl.mirror1.clear()
+            bl.mirror2.clear()
+            start_time = time.time()
+            n += 1
+        bl.pause(1.0)
+
+    print("Got beads!")
+
+
+def goto_distance(target, match_threshold, speed=1, tolerance=0.2):
+    """Move trap 1 until it reaches the `target` distance from trap 2.
+
+    Note: This throws an error if the beads are lost (since we will not have a reliable distance then either)"""
+    dx = target - distance.latest_value
+    throw_if_beads_lost(match_threshold)
+
+    while abs(dx) > tolerance:  # um
+        bl.mirror1.move_by(dx=0.1 if dx > 0 else -0.1, speed=speed)
+        dx = target - distance.latest_value
+        throw_if_beads_lost(match_threshold)
+
+
+def goto_force(target, match_threshold, speed=1, tolerance=1, tether_lost_threshold=5):
+    """Move trap 1 until it reaches the `target` force on trap 2.
+
+    Note: This throws an error if the beads are lost"""
+    df = target - force.latest_value
+    throw_if_beads_lost(match_threshold)
+
+    while abs(df) > tolerance:  # um
+        bl.mirror1.move_by(dx=0.05 if df > 0 else -0.05, speed=speed)
+        df = target - get_force(dt=0.2)
+        throw_if_beads_lost(match_threshold)
+    check_tether_breakage(tether_lost_threshold)
+
+
+def get_force(dt=0.5):
+    t0 = bl.timeline.now
+    bl.pause(dt)
+    t1 = bl.timeline.now
+    f = np.mean(force[t0:t1].data)
+
+    return f
+
+
+def catch_dna(min_distance, max_distance, match_threshold, force_threshold, fishing_speed, fishing_attempts, dt):
+    """Moves to the DNA channel and starts oscillating the trap until a prescribed force threshold is reached."""
+    print("Moving to DNA channel")
+    bl.microstage.move_to(name_dna_channel)
+    bl.pause(2.0)
+    print("reset force")
+    bl.reset_force()
+    f_current = get_force(dt=dt)
+
+    attempts = 0
+    max_retries = fishing_attempts*fishing_attempts
+    while f_current < force_threshold:
+        if attempts > max_retries:
+            raise RuntimeError(f"Max DNA fishing attempts {max_retries} reached.")
+
+        if attempts % fishing_attempts == 0:  # If the number of attempts is an integer times max_retries, do this:
+            print("Moving to DNA channel")
+            bl.microstage.move_to(name_dna_channel)
+            print(f"Moving beads a distance {dna_length} um apart")

```

Comment:
One thing I find a little strange is that you obtain some of the parameters from the global state (`dna_length`) and some from input parameters `min/max distance`. I would only use the input parameters here, to avoid having them go out of sync.